web/ui/static/js/graph/index.js

```diff
@@ -260,10 +260,11 @@ Prometheus.Graph.prototype.populateInsertableMetrics = function() {
 };
 
 Prometheus.Graph.prototype.initTypeahead = function(self) {
-  const historyIsChecked = $("div.query-history").hasClass("is-checked");
-  const source = historyIsChecked ? pageConfig.allMetrics.concat(JSON.parse(localStorage.getItem("history"))) : pageConfig.allMetrics;
+  const source = queryHistory.isEnabled() ? pageConfig.queryHistMetrics.concat(pageConfig.allMetrics) : pageConfig.allMetrics;
 
   self.expr.typeahead({
+    afterSelect: this.submitQuery.bind(this),
```

Comment:
Just a thought, in master prometheus you have to click "enter" twice right now to do a query from the typeahead (once to select an entry in the typeahead and once to submit the query). Now the query would be submitted on the first enter if something in the typeahead is selected, but would still need two enters if nothing is selected in the typeahead. 

The difference in behavior could feel a bit buggy, but I am not sure on the correct answer since one enter queries are nice.

Reply:
Good catch. At least when selecting just a normal metric from the dropdown, I normally wouldn't want it to evaluate the expression immediately, since usually you'll want to add other expression bits around it first. For the case of selecting an entire historical expression, I think either is fine, but maybe it's better to be consistent and not immediately evaluate in either case.